eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/RefundsAndDeposits.hs

```diff
@@ -81,17 +88,18 @@ keyCertsRefunds ::
   DPState c ->
   [DCert c] ->
   Coin
-keyCertsRefunds pp dpstate certs = snd (foldl' accum (regkeys, Coin 0) certs)
+keyCertsRefunds pp dpstate certs = snd (foldl' accum (initialKeys, Coin 0) certs)
   where
-    regkeys = dsDeposits (dpsDState dpstate)
+    initialKeys = (Rewards . dsUnified . dpsDState) dpstate
+    keyDeposit = compactCoinOrError (getField @"_keyDeposit" pp)
     accum (keys, ans) (DCertDeleg (RegKey k)) =
       -- Deposit is added locally to the growing 'keys'
-      (Map.insert k (getField @"_keyDeposit" pp) keys, ans)
+      (Rewards $ UM.insert k (RDPair mempty keyDeposit) keys, ans)
     accum (keys, ans) (DCertDeleg (DeRegKey k)) =
```

Comment:
Unrelated to your changes, just an observation. It would be good to make the tuple acc strict: `(!keys, !ans)` in both positions (line 95 and line 98).

Reply:
Good catch, done